### ISSUE TITLE
Create a new buffer if needed when closing Niffler

### ISSUE THREAD

**Issue by marlun** [pgdouyon/vim-niffler#20]
Another small change. If I open vim, search for a file with Niffler but decide to close it then the Niffler window is not closed. I feel that the user would except to get back to the empty buffer vim created at start?

**Comment by marlun** [comment]
I thought something like this would fix it

``` diff
diff --git a/plugin/niffler.vim b/plugin/niffler.vim
index 0bd859e..0ae7b3d 100644
--- a/plugin/niffler.vim
+++ b/plugin/niffler.vim
@@ -410,7 +410,11 @@ function! s:quit_niffler(prompt)
     let save_wd = b:niffler_save_wd
     call matchdelete(b:niffler_highlight_group)
     call setmatches(b:niffler_save_matches)
-    execute "keepalt keepjumps buffer ".b:niffler_origin_buffer
+       if b:niffler_origin_buffer
+           execute "keepalt keepjumps buffer ".b:niffler_origin_buffer
+       else
+           enew
+       endif
     execute "lchdir! " . save_wd
 endfunction
```

But then I thought of what would happen if you start vim, write something in the new buffer which vim creates at start and then open niffler without saving. With the above code `:enew` will be called instead of going back to the unsaved buffer. I'm not sure how to check for the unsaved buffer yet.

Niffler without the above patch will not handle the unsaved buffer either, it just stay open.

**Comment by pgdouyon** [comment]
Vim is weird, had to add some weird redraws to get this to work well.

I just deleted the Niffler buffer instead of creating a new one because this function is also called when making a selection, so we would be unnecessarily creating a new buffer in that case if there was no origin buffer.  Let me know if this fixes it for you.

I also renamed the function to `close_niffler` instead of `quit_niffler` to avoid that confusion, otherwise it's hard to remember that this function is actually called in other places (i.e. opening the selection).
